File: java-app/src/main/java/com/android/developer/tools/MyApplication.java
package com.android.developer.tools;

import android.app.Application;


/**
 * @author tianlong
 * @date 2017/12/18.
 */
public class MyApplication extends Application {
    public static MyApplication instance;

    @Override
    public void onCreate() {
        super.onCreate();
        instance = this;
    }
}

File: kotlin-app/src/main/java/com/android/developer/tools/MyApplication.kt
package com.android.developer.tools

import android.app.Application

/**
 * @author tianlong
 * @date  2017/12/18.
 */
class MyApplication : Application() {

    companion object {
        lateinit var instance: MyApplication
    }

    override fun onCreate() {
        super.onCreate()
        instance = this
    }
}